SU2_CFD/src/driver_adjoint_singlezone.cpp

```diff
@@ -0,0 +1,521 @@
+
+    SetAdj_ObjFunction();
+
+    /*--- Interpret the stored information by calling the corresponding routine of the AD tool. ---*/
+
+    AD::ComputeAdjoint();
+
+    /*--- Extract the computed adjoint values of the input variables and store them for the next iteration. ---*/
+
+    iteration->Iterate(output, integration_container, geometry_container,
+                         solver_container, numerics_container, config_container,
+                         surface_movement, grid_movement, FFDBox, ZONE_0, INST_0);
+
+    /*--- Monitor the pseudo-time ---*/
+    StopCalc = iteration->Monitor(output, integration_container, geometry_container,
+                                  solver_container, numerics_container, config_container,
+                                  surface_movement, grid_movement, FFDBox, ZONE_0, INST_0);
+
+    /*--- Clear the stored adjoint information to be ready for a new evaluation. ---*/
+
+    AD::ClearAdjoints();
+
+    if (config->GetTime_Domain())
+      output->SetConvHistory_Body(NULL, geometry_container, solver_container, config_container, integration_container, true, 0.0, ZONE_0, INST_0);
+
+    if (StopCalc) break;
+
+  }
+
+}
+
+void CDiscAdjSinglezoneDriver::Postprocess() {
+
+
+
+  if (config->GetKind_Solver() == DISC_ADJ_EULER ||
+      config->GetKind_Solver() == DISC_ADJ_NAVIER_STOKES ||
+      config->GetKind_Solver() == DISC_ADJ_RANS){
+
+    /*--- Compute the geometrical sensitivities ---*/
+    SecondaryRecording();
+
+  }
+
+  if (config->GetKind_Solver() == DISC_ADJ_FEM){
+
+    /*--- Apply the boundary condition to clamped nodes ---*/
+    iteration->Postprocess(output,integration_container,geometry_container,solver_container,numerics_container,config_container,surface_movement,grid_movement,FFDBox,ZONE_0,INST_0);
+
+  }
+
+}
+
+void CDiscAdjSinglezoneDriver::SetRecording(unsigned short kind_recording){
+
+  AD::Reset();
+
+  /*--- Prepare for recording by resetting the solution to the initial converged solution---*/
+
+  iteration->SetRecording(solver_container, geometry_container, config_container, ZONE_0, INST_0, kind_recording);
+
+  /*---Enable recording and register input of the iteration --- */
+
+  if (kind_recording != NONE){
+
+    AD::StartRecording();
+
+    if (rank == MASTER_NODE && kind_recording == MainVariables) {
+      cout << endl << "-------------------------------------------------------------------------" << endl;
+      cout << "Direct iteration to store the primal computational graph." << endl;
+      cout << "Compute residuals to check the convergence of the direct problem." << endl;
+    }
+    iteration->RegisterInput(solver_container, geometry_container, config_container, ZONE_0, INST_0, kind_recording);
+
+  }
+
+  /*--- Set the dependencies of the iteration ---*/
+
+  iteration->SetDependencies(solver_container, geometry_container, numerics_container, config_container, ZONE_0, INST_0, kind_recording);
```

Comment:
I have a request, can you break some of these lines? I find myself horizontal scrolling too many times, maybe 120 characters long?

Reply:
Sure, just limited the lines to 118 characters, which I think it's the limit of the github editor. Will push it in a bit.